### PR TITLE
deploy-qemu: rewrite in python

### PR DESCRIPTION
**Pull request by msehnout**
The discussion about Bash syntax have reached the threshold where it is
necessary to rewrite the script in Python to make everyone's life
easier.

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: tools/deploy-qemu.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+if len(sys.argv) != 3:
+    print(HELP_TEXT)
+    exit(1)
+
+script_dir = os.path.dirname(os.path.realpath(__file__))
+image = sys.argv[1]
+userdata = sys.argv[2]
+
+qemu_extra_args = os.getenv("QEMU_EXTRA_ARGS")
+qemu_extra_args = qemu_extra_args.split(' ') if qemu_extra_args is not None else []
```
Comment (larskarlitski) [inline tools/deploy-qemu.py:32]: I think you can drop this line and modify the previous one to:

```python
qemu_extra_args = os.getenv("QEMU_EXTRA_ARGS", [])
```

File: tools/deploy-qemu.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    subprocess.run(["qemu-system-x86_64",
+        "-M", "accel=kvm:hvf",
+        "-m", "1024",
+        "-snapshot",
+        "-cpu", "host",
+        "-net", "nic,model=virtio",
+        "-net", "user,hostfwd=tcp::2222-:22,hostfwd=tcp::4430-:443",
+        "-cdrom", f"{workdir}/cloudinit.iso"] +
+        qemu_extra_args +
```
Comment (larskarlitski) [inline tools/deploy-qemu.py:70]: I prefer think I'd have preferred the inline `*` syntax here, but it's a matter of taste:
```python3
["qemu-system-x86_64", ..., *qemu_extra_args, image]
```

File: tools/deploy-qemu.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+import os
```
Comment (larskarlitski) [inline tools/deploy-qemu.py:1]: A shebang would be nice.
```
#!/usr/bin/env python3
```

**Review by larskarlitski — CHANGES_REQUESTED** [review]
Very nice! Less bash is always good. Thanks ❤️ 

Works as advertised, but please drop the `.py` extension so that the documentation stays up to date and people (ehm, me) aren't confused :)

File: tools/deploy-qemu.py (hunk 1 above)
Comment (msehnout) [reply tools/deploy-qemu.py:32]: It actually is more complicated. If you simply read the environment variable you could get sth like "-smp 2 -m 2048", but you need to split it to make it a list.

I've considered these two options, but rejected both:
 * This returns either a string or a list, so the processing will be difficult: `os.getenv("QEMU_EXTRA_ARGS", [])`
 * `os.getenv("QEMU_EXTRA_ARGS", "").split(' ')` returns `[""]` when QEMU_EXTRA_ARGS is empty

Eventually I decided to go with the solution in this patch.

File: tools/deploy-qemu.py (hunk 2 above)
Comment (msehnout) [reply tools/deploy-qemu.py]: Much better, I forgot this syntax exists 👍

File: tools/deploy-qemu.py (hunk 3 above)
Comment (msehnout) [reply tools/deploy-qemu.py]: Sure, but which one?!

FTR: we agreed `/usr/bin/env` is good enough for this kind of script.

File: tools/deploy-qemu.py (hunk 1 above)
Comment (larskarlitski) [reply tools/deploy-qemu.py]: Oh dear, didn't read it properly. Thanks, makes sense!

Comment (larskarlitski) [reply tools/deploy-qemu.py]: Technically, we'd probably have to use `shlex` here, but we didn't do this in the bash version either.

**Review by larskarlitski — APPROVED** [review]

File: tools/deploy-qemu.py (hunk 1 above)
Comment (msehnout) [reply tools/deploy-qemu.py]: Oh wow, I had no idea `shlex` exists. But I hope it is just fine like this.

**Review by ondrejbudai — APPROVED** [review]
Tested on Fedora 34, works like a charm, thanks a lot for doing this!

**Comment by msehnout** [comment]
Thanks for the reviews 👍 . CI haven't finished even though it has been more than 12 hours, so I'm going to merge the PR despite it being yellow. In doesn't matter anyway because the deploy-qemu script is not executed anywhere in the CI pipeline.

**Pull request merged**